main.cpp

```diff
@@ -4,21 +4,36 @@ static bool get_best_display_mode(CGDirectDisplayID display, size_t width, size_
 static void print_display_modes(CGDirectDisplayID display);
 static bool switch_display_mode(CGDirectDisplayID display, CGDisplayModeRef mode);
 
+static void print_usage(const char *program_name);
+
 int main(int argc, const char *argv[]) {
   if (argc == 1) {
     print_display_modes(kCGDirectMainDisplay);
     return 0;
   }
 
+  if (argc < 3 || argc > 4) {
+    print_usage(argv[0]);
+    return 1;
+  }
+
+  bool has_error = false;
+  int width_index = 1;
+  int height_index = 2;
+  if (argc == 4) {
+    if (*argv[2] != 'x') {
```

Comment:
argv[2][0] is arguably clearer in this particular case. Also this would match `xRandomJunk`, if you care, `strcmp` is probably better.

Reply:
done